arch/x86/entry.S

```diff
@@ -168,6 +168,9 @@ GLOBAL(end_interrupt_handlers)
 
 .align PAGE_SIZE
 ENTRY(syscall_exit)
+    /* Restore default stack segment selector (syscalls can change it) */
+    mov $__KERN_DS, %_ASM_AX
+    mov %_ASM_AX, %ss
```

Comment:
I don't see when this can happen. The sycall/sysenter should transparently set the %ss to kernel's... Is it for some other kind of exit ?

Reply:
You're right. This commit should be the other way round. Will fix.